### PR TITLE
Decouple CONVERT(..USING..) from CONVERT()

### PR DESCRIPTION
**Pull request by bbeaudreault**
These 2 functions have different meanings, so should not be normalized to CONVERT()

https://dev.mysql.com/doc/refman/5.7/en/cast-functions.html#function_convert

@sougou @acharis

### CODE REVIEW

File: go/vt/sqlparser/ast.go [hunk 1]
```diff
@@ -1711,13 +1711,20 @@ func (node *ValuesFuncExpr) WalkSubtree(visit Visit) error {
 // ConvertExpr represents a call to CONVERT(expr, type)
 // CONVERT(expr USING transcoding_name) it not supported
 type ConvertExpr struct {
```
Comment (sougou) [inline go/vt/sqlparser/ast.go]: How about going for a completely new `ConvertUsing` type ? I think that will be less confusing.

Comment (bbeaudreault) [inline go/vt/sqlparser/ast.go]: Yea, I thought about that. I'll do that instead.

**Comment by bbeaudreault** [comment]
Fixed

**Comment by sougou** [comment]
LGTM

**Pull request merged**